api/src/main/java/org/eclipse/microprofile/reactive/streams/operators/spi/ReactiveStreamsFactoryResolver.java

```diff
@@ -77,17 +79,28 @@ private static ReactiveStreamsFactory loadFromSpi(ClassLoader cl) {
         }
 
         if (instance == null) {
-            ServiceLoader<ReactiveStreamsFactory> sl = ServiceLoader.load(
-                    ReactiveStreamsFactory.class, cl);
-            for (ReactiveStreamsFactory spi : sl) {
-                if (instance != null) {
-                    throw new IllegalStateException(
-                            "Multiple ReactiveStreamsFactory implementations found: "
-                                    + spi.getClass().getName() + " and "
-                                    + instance.getClass().getName());
-                } else {
-                    instance = spi;
+            try {
+                AccessController.doPrivileged((PrivilegedExceptionAction<Void>) () -> {
+                    ServiceLoader<ReactiveStreamsFactory> sl = ServiceLoader.load(
+                            ReactiveStreamsFactory.class, cl);
+                    for (ReactiveStreamsFactory spi : sl) {
+                        if (instance != null) {
+                            throw new IllegalStateException(
+                                    "Multiple ReactiveStreamsFactory implementations found: "
+                                            + spi.getClass().getName() + " and "
+                                            + instance.getClass().getName());
+                        } else {
+                            instance = spi;
+                        }
+                    }
+                    return null;
+                });
+            } catch (PrivilegedActionException e) {
+                Throwable t = e.getCause();
+                if (t instanceof RuntimeException) {
+                    throw new RuntimeException(t);
                 }
+                throw new RuntimeException(t);
```

Comment:
At the moment this `if` block does the same thing whether the condition is true or not.
```suggestion
                if (t instanceof RuntimeException) {
                    throw (RuntimeException) t;
                }
                if (t instanceof Error) {
                    throw (Error) t;
                }
                throw new RuntimeException(t);
```

Reply:
You're right - I was in a rush :-)

And I never realised until now that Error is unchecked!

Should be fixed now, thanks